openpype/modules/deadline/plugins/publish/submit_max_deadline.py

```diff
@@ -242,12 +248,109 @@ def _use_published_name(self, data, project_settings):
 
         return job_info, plugin_info
 
+    def get_job_info_through_camera(self, camera):
+        instance = self._instance
+        context = instance.context
+        job_info = copy.deepcopy(self.job_info)
+        exp = instance.data.get("expectedFiles")
+
+        src_filepath = context.data["currentFile"]
+        src_filename = os.path.basename(src_filepath)
+        job_info.Name = "%s - %s - %s" % (
+            src_filename, instance.name, camera)
+        for filepath in self._iter_expected_files(exp):
+            if camera not in filepath:
+                continue
+            job_info.OutputDirectory += os.path.dirname(filepath)
+            job_info.OutputFilename += os.path.basename(filepath)
+
+        return job_info
+        # set the output filepath with the relative camera
+
+    def get_plugin_info_through_camera(self, camera):
+        instance = self._instance
+        # set the target camera
+        plugin_info = copy.deepcopy(self.plugin_info)
+
+        plugin_data = {}
+        # set the output filepath with the relative camera
+        for camera_scene_path in instance.data.get("sceneFiles"):
+            if camera in camera_scene_path:
+                plugin_data["SceneFile"] = camera_scene_path
+        files = instance.data.get("expectedFiles")
+        if not files:
+            raise RuntimeError("No render elements found")
+        first_file = next(self._iter_expected_files(files))
+        old_output_dir = os.path.dirname(first_file)
+        rgb_output = RenderSettings().get_batch_render_output(camera)       # noqa
+        rgb_bname = os.path.basename(rgb_output)
+        dir = os.path.dirname(first_file)
+        beauty_name = f"{dir}/{rgb_bname}"
+        beauty_name = beauty_name.replace("\\", "/")
+        plugin_info["RenderOutput"] = beauty_name
+        renderer_class = get_current_renderer()
+
+        renderer = str(renderer_class).split(":")[0]
+        if renderer in [
+            "ART_Renderer",
+            "Redshift_Renderer",
+            "V_Ray_6_Hotfix_3",
+            "V_Ray_GPU_6_Hotfix_3",
+            "Default_Scanline_Renderer",
+            "Quicksilver_Hardware_Renderer",
+        ]:
+            render_elem_list = RenderSettings().get_batch_render_elements(
+                instance.name, old_output_dir, camera
+            )
+            for i, element in enumerate(render_elem_list):
+                if camera in element:
+                    elem_bname = os.path.basename(element)
+                    new_elem = f"{dir}/{elem_bname}"
+                    new_elem = new_elem.replace("/", "\\")
+                    plugin_info["RenderElementOutputFilename%d" % i] = new_elem   # noqa
+
+        if camera:
+            # set the default camera and target camera
+            # (weird parameters from max)
```

Comment:
huh, weird indeed.

Reply:
you need to set both camera0 cand camera1 to get the correct camera